waspc/src/Wasp/Analyzer/Parser/Tokenizer.hs

```diff
@@ -0,0 +1,26 @@
+module Wasp.Analyzer.Parser.Tokenizer
+  ( isValidWaspIdentifier,
+  )
+where
+
+import Control.Monad.Except (runExcept)
+import Control.Monad.State (evalStateT)
+import Wasp.Analyzer.Parser.Lexer (alexScanTokens)
+import Wasp.Analyzer.Parser.Monad (Parser, makeInitialState)
+import Wasp.Analyzer.Parser.Token (Token (Token), TokenType (TIdentifier))
+
+isValidWaspIdentifier :: String -> Bool
+isValidWaspIdentifier str = case tokenize str of
```

Comment:
str -> source

Reply:
I forgot to comment on this one.

I think `str` might be a better fit here. We are answering the question "is this string a valid identifier" (as opposed to "is this source a valid identifier".

Reply:
Ok, sounds good! It was part of Lexer, so then it made sense to me that input is source. `str` gives no info, since it just describes its type. `source` gives us a bit of an idea where it is coming from. But maybe `source` is assuming too much? Could be!

Reply:
> source gives us a bit of an idea where it is coming from

Yes, that's the thing. It almost certainly isn't coming from the source (i.e., it's very unlikely the caller code would call this argument source). In our only current case, we use it to check the project's name (which hardly qualifies as source).

`str` is pretty uninformative, but I'd argue that's a feature :D. It tells us that the function is simple and can operate on arbitrary small strings without particular meanings, similar to `capitalize str = ...` or `lowercase str = ...`.

Reply:
Ok, that does make sense indeed!